src/registry.rs

```diff
@@ -231,6 +240,26 @@ impl RegistryState {
         self.ready
     }
 
+    /// Returns an iterator over all globals.
+    ///
+    /// This list may be incomplete if [Self::ready] returns false, and may change later if the
+    /// compositor adds or removes globals after initial enumeration.
+    ///
+    /// No guarantees are provided about the ordering of the globals in this iterator.
+    pub fn globals(&self) -> impl Iterator<Item = Global> + '_ {
+        self.globals.iter().cloned()
```

Comment:
Why force the `.cloned()`, instead of returning `impl Iterator<Item = &Global>` ?

Reply:
I think I out-thought myself trying to plan for future changes to a Map<String, (version, name)> - but you could just as easily have a Set<Global> for that.